src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -342,17 +369,71 @@ localCompletionsForParsedModule pm@ParsedModule{pm_parsed_source = L _ HsModule{
 
     thisModName = ppr hsmodName
 
+    --recordCompls = localRecordSnippetProducer pm thisModName
+
+findRecordCompl' :: ParsedModule -> T.Text -> TyClDecl GhcPs -> [CompItem]
+findRecordCompl' pmod mn x = case x of
+    DataDecl {tcdLName, tcdDataDefn} -> findRecordCompl pmod mn tcdLName tcdDataDefn
+    _ -> []
+
+findRecordCompl :: ParsedModule -> T.Text -> Located (IdP GhcPs) -> HsDataDefn GhcPs -> [CompItem]
+findRecordCompl pmod mn lname dd = catMaybes name_type''
+    where
```

Comment:
this function would write quite well as a big list comprehension

Reply:
I am not sure what you mean by this. Could you elaborate a little. Or may be also point to other examples? Thanks

Reply:
I think this function could be rewritten quite nicely in the following style:
```
findRecordCompl pmod mn lname dd = 
  [ mkRecordSnippetCompItem ctxStr flds mn doc
  | ConDeclH98{..} <- dd_cons dd
  , con_arg <- con_args
  , Just name_flds <- [getFlds con_fields]
  , ...
  ] 
```

Reply:
I see. Ok. Let me try!